endless-sky.desktop

```diff
@@ -1,4 +1,5 @@
 [Desktop Entry]
+Version=1.5
```

Comment:
Since you're the resident linux guy, can you see this line causing any issues? Do any major DEs only support 1.4, and what would they do with this desktop file?

Reply:
Speaking of that - the appimage build does not yet support 1.5.

appimagetool uses `desktop-file-utils` under the hood ([ref](https://github.com/AppImage/AppImageKit/issues/87)).

The runner for the appimage build is currently still bionic. We could upgrade that to jammy, but even that would only bring us to [desktop-file-utils 0.26](https://packages.ubuntu.com/jammy/desktop-file-utils) which [only supports 1.4 of the spec](https://gitlab.freedesktop.org/xdg/desktop-file-utils/blob/master/NEWS).

Reply:
Stupid old distros with old versions of stuff. There isn't actually any problem with using newer keys, it's just the parser being uncecessarily strict about validation. It's a false positive caused by old tooling.

Reply:
Wait. It's been a while since i worked with them, but is the desktop file *inside* the appimage even used? I mean, they're just a normal self-contained binary, anything they unpack is only relevant once they have been executed.

Reply:
It might be used internally, but yeah, it's not relevant outside the AppImage.

Still, there's no reason to work around our own bugs IMO. We should ideally upgrade the tooling. I have `desktop-file-utils-0.26` on Fedora 36 which is my distro of choice, and it validates the desktop file successfully. So it seems like we need to upgrade the CI's Ubuntu version.

Reply:
That's strange, the changelog doesn't mention the 1.5 update. Are you sure this wasn't patched by the Fedora maintainers?

It might be worse than just a straight distro upgrade... We use the [linuxdeploy](https://github.com/linuxdeploy/linuxdeploy/)'s AppImage, and i have a feeling this is about the `desktop-file-utils` version bundled with that AppImage.

Reply:
The latest version of the spec is indeed 1.5 

> We use the [linuxdeploy](https://github.com/linuxdeploy/linuxdeploy/)'s AppImage, and i have a feeling this is about the desktop-file-utils version bundled with that AppImage.

Yes. See the discussion here https://github.com/AppImage/AppImageKit/issues/1171

Reply:
Thanks for finding that.

I might recommend that we switch to a less opinionated provider of AppImage tooling that supports newer versions of XDG specs. The maintainer's disinterest in doing so seems to bode ill for the prospect of sufficient flexibility to support diverse developer use cases.

Reply:
Unfortunately for us that is the official library to create AppImages, and every tool I checked that creates AppImages uses that library under the hood.

Reply:
Yeah that's kind of the problem with AppImages. The tech is fine, but the culture surrounding it has some issues IMO. Other projects have rejected or dropped AppImage support over these types of issues in the past. There was a recent high-profile example with OBS.

I'm not suggesting we should drop AppImage support, but I'm also not really happy about this perfectly functional and benign change being blocked by a CI failure false positive due to a 3rd-party dependency's intentional use of old tooling.

Reply:
Yeah not sure what else we can do about this. 

> being blocked by a CI failure false positive

It's not really CI, but the library used to create AppImages. I couldn't find an alternative and so if we want to have an AppImage we have to use AppImageTool which doesn't support the newer desktop entry.

I've also looked if it is possible to replace the desktop file after the AppImage is created, but alas that is not possible.

Reply:
Well it is a CI failure, right?The CI is red because AppImageTool complains about the changes to this desktop file. But the changes are fine and the issue is AppImageTool using a version of `desktop-file-validate` that isn't either new, or patched to behave correctly when given a desktop file with newer keys.

@probonopd how would you suggest we proceed here? We want these keys in our desktop file, and we want an AppImage. Can you help us move forward?

Reply:
Update: it looks like there's movement on that issue. And until it's formally resolved, there's a local workaround we can do: https://github.com/AppImage/AppImageKit/issues/1171#issuecomment-1160737818

Reply:
The issue was closed, but I celebrated too soon - needs one more commit to https://github.com/linuxdeploy/linuxdeploy to trigger their CI.

Reply:
Unless the CI pulls its data from the git master branch of AppImageKit, re-running the CI won't be enough; we need to get the CI updated to include the release that the fix eventually lands in.

Reply:
> Unless the CI pulls its data from the git master branch of AppImageKit

It does! And we use the latest CI run as well, so no need to wait for a release (fortunately).

Reply:
Oh ok, great! Looks like the CI is still failing though.

Reply:
Yeah we need to wait on linuxdeploy's CI to build (that CI uses the master branch of AppImageKit), either manually triggered or by a commit.

Reply:
new linuxdeploy version, new CI run, new error.

It complains about `SingleMainWindows` but *not* `PrefersNonDefaultGPU`. also this:

```
error: value "1.5" for key "Version" in group "Desktop Entry" is not a known version
```

What are the odds this version of desktop-file-validate only supports 1.4 of the spec?